File: Actividades en el lenguaje C++/Tarea_2_Declarar_Tipo_de_Dato_Abstracto.cpp
//ACTIVIDAD 2: Como declarar un tipo de dato abstracto 
//Alumno: Cristian Pabel Vidaña Herrera
//Fecha: 28/agosto/2025
//Materia: Estructura de Datos


#include <iostream>
using namespace std;

// Definición de la estructura Persona
struct Persona {
    string nombre;  // atributo nombre
    int edad;       // atributo edad
};

int main() {
    // Arreglo de estructuras Persona
    Persona personas[3] = {
        {"Ana", 25},
        {"Luis", 30},
        {"Maria", 22}
    };

    // Recorrer arreglo y mostrar datos
    for (int i = 0; i < 3; i++) {
        cout << personas[i].nombre << " " << personas[i].edad << endl;
    }

    return 0;
}

File: Actividades en el lenguaje Python/Tarea_2_Declarar_Tipo_de_Dato_Abstracto.py
#ACTIVIDAD 2: Como declarar un tipo de dato abstracto 
#Alumno: Cristian Pabel Vidaña Herrera
#Fecha: 28/agosto/2025
#Materia: Estructura de Datos

# Definición de la clase Persona
class Persona:
    def __init__(self, nombre, edad):  # constructor que recibe nombre y edad
        self.nombre = nombre           # atributo nombre
        self.edad = edad               # atributo edad

# Lista de personas (arreglo de objetos)
personas = [
    Persona("Ana", 25),
    Persona("Luis", 30),
    Persona("Maria", 22)
]

# Recorrer la lista y mostrar los datos
for p in personas:
    print(p.nombre, p.edad)  # imprime nombre y edad de cada persona

